src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelDepGraphFunction.java

```diff
@@ -181,6 +181,9 @@ private ImmutableBiMap<String, ModuleExtensionId> calculateUniqueNameForUsedExte
   private static String makeUniqueNameCandidate(ModuleExtensionId id, int attempt) {
     Preconditions.checkArgument(attempt >= 1);
     String extensionNameDisambiguator = attempt == 1 ? "" : String.valueOf(attempt);
+    // An innate extension name is of the form @repo//path/to/defs.bzl%repo_rule_name, which cannot
+    // be part of a valid repo name.
+    String extensionName = id.isInnate() ? "_repo_rules" : id.getExtensionName();
```

Comment:
it would probably help things a bit if we call this `_repo_rules_X` where X is the repo rule name.

Reply:
I tried that first, but the large number of test failures told me that this would make the change more breaking by changing all canonical repo names. Assuming that many users probably only use this for `http_archive`, the current scheme wouldn't possibly break them.

The name of the repo is also already unique since it's imported, so I felt that this isn't needed, but I'm still open to changing this. I just want to make sure you are aware of my thoughts.

Reply:
no that's fine, this definitely works as is and it wasn't a strong suggestion anyhow.